packages/mendel-atomic-css/index.js

```diff
@@ -0,0 +1,74 @@
+const vm = require('vm');
+const path = require('path');
+const m = require('module');
+const nodeModuleCache = {};
+
+function isNodeModule(filePath) {
+    return filePath.indexOf('node_modules') >= 0;
+}
+
+function runEntryInVM(entryProxy, sandbox, resolver) {
+    const filename = entryProxy.id;
+    const cacheable = isNodeModule(filename);
+    if (cacheable && nodeModuleCache[filename]) {
+        return nodeModuleCache[filename];
+    }
+
+    function localRequire(requireLiteral) {
+        // TODO remove hard coded "main"
+        const depNormId = entryProxy.deps[requireLiteral].main;
+        return runEntryInVM(resolver(depNormId), sandbox, resolver);
+    }
+
+    const exports = {};
+    const module = {exports};
+
+    // the filename is only necessary for uncaught exception reports to point to the right file
+    const nodeSource = vm.runInContext(m.wrap(entryProxy.source), sandbox, {filename});
+    // function (exports, require, module, __filename, __dirname)
+    nodeSource(exports, localRequire, module, filename, path.dirname(filename));
+
+    if (cacheable) {
+        nodeModuleCache[filename] = module.exports;
+    }
+
+    return module.exports;
+}
+
+function evalEntries(entries, mainEntry) {
+    const sandbox = {console};
+    vm.createContext(sandbox);
+
+    const entryMap = new Map();
+    entries.forEach(entry => entryMap.set(entry.normalizedId, entry));
+
+    return runEntryInVM(mainEntry, sandbox, (id) => entryMap.get(id));
+}
+
+module.exports = {
+    kind: 'gst',
+    predicate: function(entry) {
+        return !isNodeModule(entry.id) && entry.source.indexOf('getAtomicClasses') >= 0;
+    },
+    transform: function(context, entries) {
```

Comment:
For the sake of comparison, here is a snippet of a `mendel-transform` API:

```js
function({source, filename}, options) {
  /*__*/
  return {source, map};
}
```

How about we make GST API very similar?

```js
function({source, filename, deps}, options, context) {
  /*__*/
  return {source, map, deps};
}
```

The above suggestion would not change any principle. Just make it look similar.
But if you are willing to consider larger changes, I would go further and do something like:

```js
function({source, filename, deps}, options) {
  /*__*/
  return {source, map};
}
```

I think from our previous conversation, that you got the impression that I am just avoiding work. It is not the case, I think it is about making things better. Here is a bit of reasoning:

  1. Reduced API surface surface feels nice. `addVirtualEntry` pattern is not something I am super excited about, even in potential final form. I think it will be a hard concept to convey in documentation, requires too much understanding of Mendel internals and having and API like that is supporting edge cases that will happen in the future that we don't even foresee, people will use in many weird ways.
  2. `deps` in the return value is almost a side effect. It opens a can of worms: should we honor even if we don't find the dep? should we actually check? should we keep the previous values? are those deps just the direct dependencies or should it be the array with the whole graph of descendants (that's the input deps, so why output is objec)?

This would be so close to regular transform. Looks better, feels better, easy to explain: "If you need to look at your deps, you do GST, if not, it is IST." or even better, instead of `{type, predicate, transform}` it could be `{waitsForDeps=<Boolean>, predicate, transform}`.

Just think about it. We can leave as is, I am ok with it and I do understand the potential of future additions to context you are envisioning. I just wanted to make sure I give you the other side of the argument in full with a bit of better reasoning than our previous conversations.